src/main/java/com/cst438/service/GradebookServiceProxy.java

```diff
@@ -0,0 +1,254 @@
+package com.cst438.service;
+
+import com.cst438.dto.*;
+import com.cst438.domain.*;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+public class GradebookServiceProxy {
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Autowired
+    private SectionRepository sectionRepository;
+
+    @Autowired
+    private EnrollmentRepository enrollmentRepository;
+
+    @Autowired
+    private TermRepository termRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    Queue gradebookServiceQueue = new Queue("gradebook_service", true);
+
+    @Bean
+    public Queue createQueue() {
+        return new Queue("registrar_service", true);
+    }
+
+    @Autowired
+    RabbitTemplate rabbitTemplate;
+
+    @RabbitListener(queues = "registrar_service")
+    public void receiveFromRegistrar(String message) {
+        try {
+            MessageWrapper wrapper = fromJsonString(message, MessageWrapper.class);
+            String action = wrapper.getAction();
+            String payload = wrapper.getPayload();
+            switch (action) {
+                case "addCourse":
+                case "updateCourse":
+                case "deleteCourse":
+                    handleCourseAction(action, payload);
+                    break;
+                case "addSection":
+                case "updateSection":
+                case "deleteSection":
+                    handleSectionAction(action, payload);
+                    break;
+                case "addUser":
+                case "updateUser":
+                case "deleteUser":
+                    handleUserAction(action, payload);
+                    break;
+                case "addEnrollment":
+                case "deleteEnrollment":
+                    handleEnrollmentAction(action, payload);
+                    break;
+                case "updateFinalGrade":
```

Comment:
This is the only action that this method needs to handle. All of the add/update/delete functions are handled on the other service.